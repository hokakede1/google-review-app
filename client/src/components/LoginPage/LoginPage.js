import React, { Component } from "react";
import { Segment } from "semantic-ui-react";
import Radium from "radium";

const styles = {
  wrapper: {
    height: "100vh",
    width: "100%",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    // background: 'linear-gradient(#ff9a9e,#fecfef)'
    background:
      "linear-gradient(rgba(43, 44, 78, 0.8), rgba(43, 44, 78, 1)), url(https://source.unsplash.com/cNgsAdd4-m4)",
    backgroundSize: "cover",
    backgroundRepeat: "no-repeat"
  },
//   loginSegment: {
//     width: "100%",
//     height: "250px",
//     "@media (min-width: 500px)": {
//       width: "400px",
//       height: "300px"
//     }
//   },
  loginContainer: {
    color: "#B2B2B2",
    fontSize: "14px",
    fontWeight: "400"
  },
  signInUpContainer: {
    width: "180px",
	marginBottom: "120px",
	marginLeft: "12px",
  },
  userSelected: {
	  color: "#E3E3E3",
	  fontSize: "21px",
	  marginBottom: "5px",
  },
  userUnselected: {
	color: "#717595",
	fontSize: "21px",
	marginBottom: "5px",
  },
  loginHeader: {
    marginLeft: "12px"
  },
  loginInput: {
    width: "320px",
    height: "35px",
    marginTop: "10px",
    marginBottom: "15px",
    paddingLeft: "16px",
    backgroundColor: "rgba(113, 117, 149, 0.2)",
    border: "0",
    borderRadius: "25px",
    fontSize: "15px",
    color: "#E3E3E3"
  },
  loginButton: {
    backgroundColor: "#1762EE",
    width: "320px",
    height: "35px",
    borderRadius: "25px",
    color: "#E3E3E3",
    border: "none",
    marginBottom: "90px"
  },
  keepSignedInContainer: {
    display: "flex",
    marginLeft: "12px",
    marginBottom: "15px"
  },
  keepSignedInCheckbox: {

  },
  underlineStyle: {
    height: "2px",
    color: "#717595",
    background: "rgba(113, 117, 149, 0.2)",
    fontSize: "0",
    border: "0"
  },
  forgotPasswordContainer: {
	textAlign: "center",
	marginTop: "51px",
	color: "#717595",
  }, 
  displayFlex: {
    display: "flex",
    justifyContent: "space-between"
  }
};

class LoginPage extends Component {
  state = {
    signIn: true,
    signUp: false
  };

  handleSignIn = () => {
    this.setState({
      signIn: true,
      signUp: false
    });
  };

  handleSignUp = () => {
    this.setState({
      signIn: false,
      signUp: true
    });
  };

  render() {
    console.log(`[STATE]`, this.state);
    return (
      <div style={styles.wrapper}>
        <div style={styles.loginContainer}>
          <div style={[styles.displayFlex, styles.signInUpContainer]}>
            <div onClick={this.handleSignIn}>
              {this.state.signIn ? (
                <div>
                  <h1 style={styles.userSelected}>SIGN IN</h1>
                  <hr
                    style={[
                      styles.underlineStyle,
                      { color: "#1762EE", background: "#1762EE", height: "2px" }
                    ]}
                  />
                </div>
              ) : (
                <div>
                  <h1 style={styles.userUnselected}>SIGN IN</h1>
                </div>
              )}
            </div>
            <div onClick={this.handleSignUp}>
			{this.state.signUp ? (
                <div>
                  <h1 style={styles.userSelected}>SIGN UP</h1>
                  <hr
                    style={[
                      styles.underlineStyle,
                      { color: "#1762EE", background: "#1762EE", height: "2px" }
                    ]}
                  />
                </div>
              ) : (
                <div>
                  <h1 style={styles.userUnselected}>SIGN UP</h1>
                </div>
              )}
            </div>
          </div>
          <div>
            <h3 style={styles.loginHeader}>USERNAME</h3>
            <input type="text" style={styles.loginInput} />
          </div>
          <div>
            <h3 style={styles.loginHeader}>PASSWORD</h3>
            <input type="text" style={styles.loginInput} />
          </div>
          <div style={styles.keepSignedInContainer}>
            <input type="checkbox" style={{marginRight: "5px"}}/>
            <h3>Keep me Signed in</h3>
          </div>
          <button style={styles.loginButton}>SIGN IN</button>
          <div>
            <hr style={styles.underlineStyle} />
          </div>
          <div style={styles.forgotPasswordContainer}>
            <h3>Forgot Password?</h3>
          </div>
        </div>
      </div>
    );
  }
}

export default Radium(LoginPage);
