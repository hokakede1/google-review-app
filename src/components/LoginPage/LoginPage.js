import React, { Component } from 'react';
import { Segment } from 'semantic-ui-react';
import Radium from 'radium';

const styles = {
  wrapper: {
    height: '100vh',
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    background: 'linear-gradient(#ff9a9e,#fecfef)',
  },
  loginSegment: {
    width: '100%',
    height: '250px',
    '@media (min-width: 500px)': {
      width: '400px',
      height: '300px',
    },
  },

};
class LoginPage extends Component {
  constructor() {
    super();
    this.state = {
      name: 'hi',
    };
  }
  render() {
    return (
      <div style={styles.wrapper}>
        <Segment raised style={styles.loginSegment}>
          <h1>{this.state.name}</h1>
        </Segment>
      </div>
    );
  }
}
// LoginPage = Radium(LoginPage)
export default Radium(LoginPage);
