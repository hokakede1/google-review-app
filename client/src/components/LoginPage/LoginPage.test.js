import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LoginPage from './LoginPage';

configure({adapter: new Adapter()});

describe('<LoginPage />', () => {
    it('should render something', () => {
        const wrapper = shallow(<LoginPage />);
        expect(wrapper.contains(<h1></h1>)).toEqual(true);
    })
})