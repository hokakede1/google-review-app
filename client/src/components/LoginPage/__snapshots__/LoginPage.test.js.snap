// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Segment
        raised={true}
        style={
                Object {
                        "@media (min-width: 500px)": Object {
                          "height": "300px",
                          "width": "400px",
                        },
                        "height": "250px",
                        "width": "100%",
                      }
        }
>
        <h1>
                hi
        </h1>
</Segment>,
      "data-radium": true,
      "style": Object {
        "alignItems": "center",
        "background": "linear-gradient(#ff9a9e,#fecfef)",
        "display": "flex",
        "height": "100vh",
        "justifyContent": "center",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <h1>
          hi
</h1>,
        "raised": true,
        "style": Object {
          "@media (min-width: 500px)": Object {
            "height": "300px",
            "width": "400px",
          },
          "height": "250px",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hi",
        },
        "ref": null,
        "rendered": "hi",
        "type": "h1",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Segment
          raised={true}
          style={
                    Object {
                              "@media (min-width: 500px)": Object {
                                "height": "300px",
                                "width": "400px",
                              },
                              "height": "250px",
                              "width": "100%",
                            }
          }
>
          <h1>
                    hi
          </h1>
</Segment>,
        "data-radium": true,
        "style": Object {
          "alignItems": "center",
          "background": "linear-gradient(#ff9a9e,#fecfef)",
          "display": "flex",
          "height": "100vh",
          "justifyContent": "center",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h1>
            hi
</h1>,
          "raised": true,
          "style": Object {
            "@media (min-width: 500px)": Object {
              "height": "300px",
              "width": "400px",
            },
            "height": "250px",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hi",
          },
          "ref": null,
          "rendered": "hi",
          "type": "h1",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
